Where_Is_My_Wife/Assets/_Main/Scripts/Input/InputEventManager.cs

```diff
@@ -0,0 +1,246 @@
+using System;
+using UniRx;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using WIMW.Input;
+using Zenject;
+
+namespace WhereIsMyWife.Managers
+{
+    public partial class InputEventManager : IPlayerInputEvent
+    {
+        private readonly Subject<Unit> _jumpStartSubject = new Subject<Unit>();
+        private readonly Subject<Unit> _jumpEndSubject = new Subject<Unit>();
+        private readonly Subject<float> _runSubject = new Subject<float>();
+        private readonly Subject<Vector2> _dashSubject = new Subject<Vector2>();
+        private readonly Subject<Vector2> _useItemSubject = new Subject<Vector2>();
+        private readonly Subject<Unit> _hookStartSubject = new Subject<Unit>();
+        private readonly Subject<Unit> _hookEndSubject = new Subject<Unit>();
+        private readonly Subject<Unit> _lookUpSubject = new Subject<Unit>();
+        private readonly Subject<bool> _lookDownSubject = new Subject<bool>();
+
+        public IObservable<Unit> JumpStartAction => _jumpStartSubject.AsObservable();
+        public IObservable<Unit> JumpEndAction => _jumpEndSubject.AsObservable();
+        public IObservable<float> RunAction => _runSubject.AsObservable();
+        public IObservable<Vector2> DashAction => _dashSubject.AsObservable();
+        public IObservable<Vector2> UseItemAction => _useItemSubject.AsObservable();
+        public IObservable<Unit> HookStartAction => _hookStartSubject.AsObservable();
+        public IObservable<Unit> HookEndAction => _hookEndSubject.AsObservable();
+        public IObservable<Unit> LookUpAction => _lookUpSubject.AsObservable();
+        public IObservable<bool> LookDownAction => _lookDownSubject.AsObservable();
+    }
+
+    public partial class InputEventManager : IInitializable
+    {
+        private PlayerInputActions _playerInputActions = new PlayerInputActions();
+        
+        private Vector2 _moveVector = Vector2.zero;
+        
+        private float _horizontalDeadZone = 0.5f;
+        
+        public void Initialize()
+        {
+            _playerInputActions.Enable();
+            SubscribeToInputActions();
+            
+            CheckForCurrentController();
+        }
+
+        private void SubscribeToInputActions()
```

Comment:
You need to unsubscribe in `OnDestroy`

Reply:
ab5628d8d62fa3bbdc7b4c2c250b8e94a0685ee0